src/blob_file_builder.cc

```diff
@@ -33,19 +33,35 @@ BlobFileBuilder::BlobFileBuilder(const TitanDBOptions& db_options,
     return;
 #endif
   }
+  block_size_ = cf_options.enable_punch_hole_gc ? cf_options.block_size : 0;
```

Comment:
I think we should sanitize the block_size in case of it is not divided by disk block size.